### DISCUSSION TITLE
ValidationError with Custom DateType

### DISCUSSION THREAD

**Discussion by sidzan** [mikro-orm/mikro-orm#2244] [Q&A]
**Describe the bug**
The production build is not able to initalize and giving issue after using customType of DateType

P.S i  do not get any issue on development, using nodemon, but while using production i am running into this issue

For production build: 
```
    "build": "babel src --out-dir build --copy-files --source-maps --extensions '.js,.ts'",
```

the following is the model i am using 
```
@Entity({ tableName: 'bookings' })
export default class Booking {
  @PrimaryKey()
  readonly id: string = ulid();
  
  ...

  @Property({
    type: DateType,
    nullable: false,
  })
  readonly scheduledAt: string;
  
}

```

**Stack trace**
```
(node:53693) UnhandledPromiseRejectionWarning: ValidationError: Could not convert JS value 'customType' of type 'string' to type DateType
at Function.invalidType (/home/username/project/node_modules/@mikro-orm/core/errors.js:61:16)
at DateType.convertToDatabaseValue (/home/username/project/node_modules/@mikro-orm/core/types/DateType.js:14:40)
at DateType.toJSON (/home/username/project/node_modules/@mikro-orm/core/types/DateType.js:33:21)
at JSON.stringify (<anonymous>)
at stringify (/home/username/project/node_modules/fast-safe-stringify/index.js:14:16)

```

**To Reproduce**
1. Create the model with a custom type of DateType
2. create a production build with typescript or babel

**Expected behavior**
The DateType should work without any issues as in the development environment

**Additional context**
I am not sure if the way i am using the Custom type is correct or not, any suggestion is much appreaciated

Babel Config as below:
```
// babel.config.js
module.exports = {
  presets: [['@babel/preset-env', { targets: { node: 'current' } }], '@babel/preset-typescript'],
  plugins: [
    ['@babel/plugin-proposal-decorators', { legacy: true }],
    ['@babel/plugin-proposal-class-properties', { legacy: true }],
    'babel-plugin-transform-typescript-metadata',
    ['module-resolver', { alias: { '@': './src' } }],
  ],
};

```

**Versions**

| Dependency | Version |
| - | - |
| node | v14.17.3 |
| typescript | ^4.3.5 |
| mikro-orm | ^4.5.9 |
| mikro-orm/postgresql | ^4.5.9 |

**Comment by B4nan** [comment]
Please provide full reproduction, I don't have any babel project I could use to test this (never really been using babel myself). 

What you could try is this:

```ts
  @Property({
    customType: new DateType(),
//    nullable: false, // not nullable is the default, no need to state it
  })
  readonly scheduledAt: string;
```

Have you seen this page? https://mikro-orm.io/docs/usage-with-babel/ 

Do you also use the `BABEL_DECORATORS_COMPAT` env var? Also I believe the order of plugins should be different (as in the docs).
